Userland/Applications/Calendar/EventManager.cpp

```diff
@@ -97,14 +97,82 @@ ErrorOr<Vector<Event>> EventManager::deserialize_events(JsonArray const& json)
     return result;
 }
 
+Core::DateTime EventManager::format_icalendar_vevent_datetime(String const& parameter)
+{
+    auto invalid_datetime = Core::DateTime::create(0);
+    if (parameter.to_byte_string().length() < 15)
+        return invalid_datetime;
+    auto formatted_string = String::formatted("{}-{}-{}T{}:{}:{}",
+        parameter.substring_from_byte_offset(0, 4).value(),
+        parameter.substring_from_byte_offset(4, 2).value(),
+        parameter.substring_from_byte_offset(6, 2).value(),
+        parameter.substring_from_byte_offset(9, 2).value(),
+        parameter.substring_from_byte_offset(11, 2).value(),
+        parameter.substring_from_byte_offset(13, 2).value());
+    if (formatted_string.is_error())
+        return invalid_datetime;
+    auto datetime = Core::DateTime::parse(DATE_FORMAT, formatted_string.value());
+    return datetime.has_value() ? datetime.value() : invalid_datetime;
+}
+
+ErrorOr<Vector<Event>> EventManager::parse_icalendar_vevents(ByteBuffer const& content)
+{
+    Event event;
+    Vector<Event> events;
+    ICalendarParserState state = ICalendarParserState::Idle;
+    StringBuilder builder;
+    TRY(builder.try_append((char const*)content.data(), content.size()));
+    auto lines = TRY(TRY(builder.to_string()).split('\n'));
+    for (size_t i = 0; i < lines.size(); i++) {
+        auto section = TRY(lines[i].split_limit(':', 2));
+        if (section.size() > 1) {
+            auto property = section[0];
```

Comment:
The RFC has `DTSTART;TZID=US-Eastern:19980119T020000` as example. I suppose we're just ignoring times with TZIDs for now? (Which is fine, just trying to understand.)

Reply:
> I suppose we're just ignoring times with TZIDs for now? (Which is fine, just trying to understand.)

For now, we can ignore the TZID parameter and keep the time value, until Calendar Events are able to handle time zones.